sycl/include/sycl/khr/free_function_commands.hpp

```diff
@@ -148,45 +148,54 @@ void launch_grouped(handler &h, range<3> r, range<3> size,
   h.parallel_for(nd_range<3>(r, size), k);
 }
 
-template <typename KernelType>
-void launch_grouped(const queue &q, range<1> r, range<1> size,
-                    const KernelType &k,
+template <
+    typename KernelType,
+    // overload of launch_grouped for sycl::kernel must be preferred
+    typename = typename std::enable_if<!std::is_same<
+        typename std::decay<KernelType>::type, sycl::kernel>::value>::type>
```

Comment:
* We have a minimum requirement of C++17, `std::*_[vt]` are available (e.g., `std::decay_t`/`std::enable_if_t` and `std::is_same_v`).
* I'd prefer SFINAE on the return type instead, but I'm not sure how consistent we are with style throughout our codebase.

Reply:
Do you mean something like "exclude sycl::kernel from overload resolution" is too trivial? Probably.